stable_baselines/common/distributions.py

```diff
@@ -354,6 +358,10 @@ def __init__(self, nvec, flat, action_mask_vector):
         self.flat = flat
         self.categoricals = list(map(CategoricalProbabilityDistribution, tf.split(flat, nvec, axis=-1)))
 
+    def set_action_mask(self, action_mask):
+        for i, categorical in enumerate(self.categoricals):
+            categorical.set_action_mask(action_mask[:, i])
+
```

Comment:
I prefer to split the dataflow(action mask) in init.

Reply:
But then that causes changing a bunch of function headers to do so, when every probability distribution class doesn't need it

Reply:
It seems that set_action_mask is required now, isn't it?
If this method is not call, the action_mask will be None.
Tf.add will get an error

From other perspectives, our approach seems to be the same.

Reply:
the sample method will not be called before set_action_mask gets executed, though. Look at the very next line

Reply:
Me too, but I think set_action_mask will increase the time complexity.